src/dependency-manager/manager.ts

```diff
@@ -44,7 +44,7 @@ export default abstract class DependencyManager {
     // Load component tasks
     for (const [task_name, task_config] of Object.entries(component.tasks)) {
       const node = new TaskNode({
-        ref: buildNodeRef(component, 'tasks', task_name),
+        ref: task_config.reserved_name || buildNodeRef(component, 'tasks', task_name),
```

Comment:
Couldn't we move this logic into `buildNodeRef`? 

something like:
`component_config[resource_type][resource_name].reserved_name`